### ISSUE TITLE
Questions about ctc_decode in tensorflow&keras

### ISSUE THREAD

**Issue by Haskely** [tensorflow/tensorflow#42488]
https://github.com/tensorflow/tensorflow/blob/b36436b087bd8e8701ef51718179037cccdfc26e/tensorflow/python/ops/ctc_ops.py#L288-L335

About this function: tf.nn.ctc_greedy_decoder
[https://www.tensorflow.org/api_docs/python/tf/nn/ctc_greedy_decoder]

I want to know how this function specifies the Class Index to be used for the BLANK LABEL
Beacuse:  
  # gen_ctc_ops.ctc_loss_v2 differs from gen_ctc_ops.ctc_loss. v2 assumes the blank index to be 0, but v1 views it as the last index.
If it doesn't know which is the BLANK LABEL, how to combine and del BLANK LABEL for the final result?
If it takes the last index as a blank label by default, there is a contradiction with the new version of the API

**Comment by Haskely** [comment]
https://github.com/tensorflow/tensorflow/blob/a195c757e3e968f6dcbb5b1ad76c95c61dc7fb67/tensorflow/core/kernels/ctc_decoder_ops.cc#L223-L224

I have found its implementation in the c language source code, and found that it does takes num_classes - 1 as the blank label by default, so this is a bug.

**Comment by Saduf2019** [comment]
@Haskely 
Please fill in the issue template for us to help you.
